include/dynd/kernels/field_access_kernel.hpp

```
@@ -6,30 +6,62 @@
-      std::cout << "y = " << *reinterpret_cast<double *>(src[0] + data_offsets[1]) << std::endl;
+      dst_tp = get_field_type(src_tp, kwds);
     }
 
-    static void instantiate(char *DYND_UNUSED(static_data), char *DYND_UNUSED(data), kernel_builder *ckb,
-                            const ndt::type &DYND_UNUSED(dst_tp), const char *DYND_UNUSED(dst_arrmeta),
-                            intptr_t DYND_UNUSED(nsrc), const ndt::type *DYND_UNUSED(src_tp),
+    static void instantiate(char *DYND_UNUSED(static_data), char *data, kernel_builder *ckb, const ndt::type &dst_tp,
+                            const char *dst_arrmeta, intptr_t DYND_UNUSED(nsrc), const ndt::type *src_tp,
                             const char *const *src_arrmeta, kernel_request_t kernreq, intptr_t DYND_UNUSED(nkwd),
-                            const array *DYND_UNUSED(kwds),
-                            const std::map<std::string, ndt::type> &DYND_UNUSED(tp_vars))
+                            const array *kwds, const std::map<std::string, ndt::type> &tp_vars)
+    {
+      const uintptr_t data_offset = get_data_offset(src_tp, src_arrmeta, kwds);
+      const ndt::type field_type[1] = {get_field_type(src_tp, kwds)};
+      const nd::array field_value = nd::empty(field_type[0]);
+      const char *field_metadata[1] = {field_value->metadata()};
+
+      ckb->emplace_back<field_access_kernel>(kernreq, data_offset);
+
+      static const array error_mode(opt<assign_error_mode>());
+      assign::get()->instantiate(assign::get()->static_data(), data, ckb, dst_tp, dst_arrmeta, 1, field_type,
+                                 field_metadata, kernreq | kernel_request_data_only, 1, &error_mode, tp_vars);
+    };
+
+    static const ndt::type &get_field_type(const ndt::type *src_tp, const array *kwds)
+    {
+      const ndt::struct_type *s = src_tp->extended<ndt::struct_type>();
+      const std::string &name = kwds[0].as<std::string>();
+      uintptr_t index = s->get_field_index(name);
```

Looking at it again, I think we need something like:
+      intptr_t index = s->get_field_index(name);
+
+      if (index == -1) {
+        throw std::invalid_argument("no field named '" + name + "'");
+      }

Or should we throw the exception in get_field_index()?